### DISCUSSION TITLE
Instructions on creating a bootable harddrive image with RPI & SHELL commands?

### DISCUSSION THREAD

**Discussion by benjamink** [tjboldt/Apple2-IO-RPi#118] [General]
I'm trying to create a hard drive image that I can boot as disk1 that includes the RPI & SHELL commands but I can't seem to get it.  I've used [AppleCommander](https://applecommander.github.io/) to create the disk from the official [ProDOS 2.4.2](https://prodos8.com/releases/prodos-242/) source disk then copied the *.bin & *.bas files into the disk from the `Apple2-IO-RPi/RaspberryPi/driveimage` directory.  When I boot with the hard drive image I created I get the error `Relocation/Configuration Error`.  Are the `PRODOS.SYS` & `BASIC.SYSTEM.SYS` patched somehow & the ones from the official ProDOS source won't work?  

How can I turn the contents of the `driveimage/` directory into a 32MB ProDOS .hdv image that I can boot to & read/write from/to?  I'm trying to have an all-in-one work HDD that I can put other stuff on too.

**Comment by tjboldt** [comment]
The PRODOS.sys and BASIC.SYSTEM.sys the official versions (but with a .sys extension added to the name). The files get dynamically imported into an in-memory drive that is generated on the fly. The other .bas files are just text files on the RPi but get dynamically converted to actual AppleSoft BASIC files during the import. There is no need to make a drive image for the boot volume as it dynamic (but ephemeral so anything saved there will go away for now until I finish the code that writes back to host). I'd recommend using drive 2 with an image as that will save back to the image. It's a standard ProDOS order hard drive image. I am about to update the code to recursively import drive 1 so you can have more files and subdirectories pulled in from the host. It will also convert PNG and JPG files into Apple II hi-res graphics files. This will allow you to run commands like "wget" on the RPi from the Apple II and display the results.

**Reply by benjamink** [reply]
Ok, that sounds great!  I think I just need to learn how ProDOS works in general too.  Thanks!
